### ISSUE TITLE
change average coherence calculation

### ISSUE THREAD

**Issue by noah10** [EmotionCognitionLab/pvs#353]
The following changes should be made to the average coherence calculation:

- Throw out the first 30 seconds of coherence values received after the start of a regime - they should not be used in calculating the average coherence. Note that "the first 30 seconds of coherence values" could be anything from 0 to 60 coherence values, depending on how reliably emWave is sending us data.
- Also ignore any artifact notifications in the first 30 seconds
- Once the regime has ended, discard the segment if it has fewer than 360 coherence values (the number we would expect for 3 minutes of data if we receive every single expected data point)
- Assuming that we have 360+ coherence values and aren't discarding the segment, use all of them to calculate the average coherence. It's fine if the different average coherence scores are based on different numbers of coherence values.

@maramather and @HyunjooY , let me know if you see anything wrong or missing here.

**Comment by maramather** [comment]
That looks good to me!

**Comment by HyunjooY** [comment]
Looks good.
One question, would this change be applied on the existing data?

**Comment by noah10** [comment]
No, it would not. We do not have the information necessary to change the data retroactively. (Once we've calculated the average coherence score we throw away all of the intermediate values that we used to calculate it.)

**Comment by HyunjooY** [comment]
Got it. 
Thank you!

**Comment by noah10** [comment]
By default the change to ignore the first 30 seconds of data (coherence values and artifacts) will apply to both rest breathing (i.e. stage 1 baseline and stage 2) as well as paced breathing. (Though the coherence score we display on-screen to the user will still update during the first 30 seconds - those values just won't be used to calculate the average coherence that we store in the database.) Is that a problem, @maramather or @HyunjooY ?

**Comment by maramather** [comment]
That is ok (I assume it is much simpler for your code).

**Comment by HyunjooY** [comment]
Yes. that is okay. Thank you!

**Comment by noah10** [comment]
Unfortunately I've just discovered that I somehow totally missed the code change necessary to drop the minimum threshold for average coherence down to 360 values from 480 values. Re-opening.
